ouroboros-network/src/Ouroboros/Network/Server/Socket.hs

```diff
@@ -0,0 +1,251 @@
+{-# LANGUAGE GADTSyntax #-}
+{-# LANGUAGE RecursiveDo #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE BangPatterns #-}
+
+-- `accept` is shadowed, but so what?
+{-# OPTIONS_GHC "-fno-warn-name-shadowing" #-}
+
+module Ouroboros.Network.Server.Socket
+  ( Accept
+  , Complete
+  , Main
+  , Stop (..)
+  , accept
+  , reject
+  , run
+
+  , Socket (..)
+  , ioSocket
+  ) where
+
+import Control.Exception (SomeException, mask, mask_, finally, onException, try)
+import Control.Concurrent.Async (Async)
+import qualified Control.Concurrent.Async as Async
+import Control.Concurrent.STM (STM)
+import qualified Control.Concurrent.STM as STM
```

Comment:
All other modules in Ouroboros.Network depend on `Control.Monad.Class.MonadSTM` instead of `Control.Concurrent.STM` directly.

Reply:
Ok. I assume the API is exactly so the same so I can just change this import and everything will work?

Reply:
The API is the same but the types are slightly different, e.g TVar IO Int instead of TVar Int.

Reply:
And we don't yet have `forkMWithUnmask` or `asyncWithUnmask` but we could if we need to.

Reply:
I think we do need those. Although I can't imagine there's any good reason to run the server in a masked state, it's still good form to try to work as well as possible in case it is.